mozetl/taar/taar_lite_guidguid.py

```diff
@@ -0,0 +1,248 @@
+            legacy addons, disabled addons, sideloaded addons.
+            """
+            return not (
+                addon.is_system or
+                addon.app_disabled or
+                addon.type != "extension" or
+                addon.user_disabled or
+                addon.foreign_install or
+                # make sure the amo_whitelist has been broadcast to worker nodes.
+                guid not in broadcast_amo_whitelist.value
+            )
+
+        # TODO: may need additional whitelisting to remove shield addons
+        # it should be populated from the current list of shield studies.
+
+        # Create an add-ons dataset un-nesting the add-on map from each
+        # user to a list of add-on GUIDs. Also filter undesired add-ons.
+        return (
+            users_df.rdd
+                .map(lambda p: (p["client_id"],
+                                [guid for guid, data in p["active_addons"].items() if is_valid_addon(guid, data)]))
+                .filter(lambda p: len(p[1]) > 1)
+                .toDF(["client_id", "addon_ids"])
+        )
+
+    logging.info("Init loading client features")
+    client_features_frame = get_initial_sample()
+    logging.info("Loaded client features")
+
+    amo_white_list = load_amo_external_whitelist()
+    logging.info("AMO White list loaded")
+
+    broadcast_amo_whitelist = sc.broadcast(amo_white_list)
+    logging.info("Broadcast AMO whitelist success")
+
+    addons_info_frame = get_addons_per_client(client_features_frame)
+    logging.info("Filtered for valid addons only.")
+
+    taar_training = addons_info_frame.join(client_features_frame, 'client_id', 'inner') \
+        .drop('active_addons') \
+        .selectExpr("addon_ids as installed_addons")
+    logging.info("JOIN completed on TAAR training data")
+
+    return taar_training
+
+
+def key_all(a):
+    """
+    Return (for each Row) a two column set of Rows that contains each individual
+    installed addon (the key_addon) as the first column and an array of guids of
+    all *other* addons that were seen co-installed with the key_addon. Excluding
+    the key_addon from the second column to avoid inflated counts in later aggregation.
+    """
+    return [(i, [b for b in a if not b is i]) for i in a]
+
+
+@click.command()
+@click.option('--date', required=True)
+@click.option('--bucket', default=OUTPUT_BUCKET)
+@click.option('--prefix', default=OUTPUT_PREFIX)
+def main(date, bucket, prefix):
+    spark = (SparkSession
+             .builder
+             .appName("taar_lite=")
+             .enableHiveSupport()
+             .getOrCreate())
+
+    logging.info("Loading telemetry sample.")
+    longitudinal_addons = load_training_from_telemetry(spark)
+
+    # Only for logging, not used, but may be interesting for later analysis.
+    guid_set_unique = longitudinal_addons.withColumn("exploded", explode(longitudinal_addons.installed_addons)).select(
```

Comment:
The sections that takes `longitudinal_addons -> addon_co_installations_collapsed` can be put into a separate transformation function. The test can then assert that the dataframe (or collected dataframe) looks right.